### PR TITLE
Make download via registry faster and less spammy

### PR DESCRIPTION
**Pull request by ingwarsw**
Should close https://github.com/earthly/earthly/issues/2351

And also its making things bit faster..
In case of one of my project tests time dropped from 66s to 53s.

### CODE REVIEW

**Comment by vladaionescu** [comment]
Very nice improvement @ingwarsw! Thanks for the contribution!

**Comment by ingwarsw** [comment]
@vladaionescu should I be worried about failing tests? Or thats because its external PR?

**Comment by vladaionescu** [comment]
Nothing to worry, we'll get this tested separately. Tests for external contributions don't work very well due to DockerHub rate limiting / login issues.

**Comment by idodod** [comment]
Hey @ingwarsw ,
Can you please look into the linting errors: https://github.com/earthly/earthly/actions/runs/6542643074/job/17766129774?pr=3393

**Comment by ingwarsw** [comment]
@idodod I was able to test it locally and its passing now..

**Review by idodod — APPROVED** [review]

**Review by alexcb — APPROVED** [review]

File: buildkitd/dockerd-wrapper.sh [hunk 1]
```diff
@@ -280,9 +284,25 @@ load_registry_images() {
...
+
+        # Wait for all background processes to finish
+        for pid in $bg_processes; do
+            wait "$pid" || {
+                echo "Downloading of images failed"
+                stop_dockerd
+                exit 1
+            }
+        done
+        end_time=$(date +%s%N | cut -b1-13)
```
Comment (alexcb) [inline buildkitd/dockerd-wrapper.sh:302]: ultra nit: What happens after 2286-11-20 17:46:39 :laughing:

answer: not our problem.

Comment (alexcb) [reply buildkitd/dockerd-wrapper.sh:302]: 9999999999 = 2286-11-20 17:46:39

**Pull request merged**